src/storage/v2/disk/storage.hpp

```diff
@@ -63,37 +63,47 @@ class DiskStorage final : public Storage {
-                                                const std::unordered_set<storage::Gid> &gids,
-                                                std::list<Delta> &index_deltas,
-                                                utils::SkipList<Vertex> *indexed_vertices,
+    void LoadVerticesFromDiskLabelPropertyIndex(const std::unordered_set<storage::Gid> &gids, LabelId label,
+                                                PropertyId property, std::list<Delta> &index_deltas,
+                                                utils::SkipList<Vertex> &indexed_vertices,
                                                 const auto &label_property_filter);
 
-    void LoadVerticesFromDiskLabelPropertyIndexWithPointValueLookup(LabelId label, PropertyId property,
-                                                                    const std::unordered_set<storage::Gid> &gids,
+    void LoadVerticesFromDiskLabelPropertyIndexWithPointValueLookup(const std::unordered_set<storage::Gid> &gids,
+                                                                    LabelId label, PropertyId property,
                                                                     const PropertyValue &value,
                                                                     std::list<Delta> &index_deltas,
-                                                                    utils::SkipList<Vertex> *indexed_vertices);
+                                                                    utils::SkipList<Vertex> &indexed_vertices);
 
     std::unordered_set<Gid> MergeVerticesFromMainCacheWithLabelPropertyIndexCacheForIntervalSearch(
         LabelId label, PropertyId property, View view, const std::optional<utils::Bound<PropertyValue>> &lower_bound,
         const std::optional<utils::Bound<PropertyValue>> &upper_bound, std::list<Delta> &index_deltas,
-        utils::SkipList<Vertex> *indexed_vertices);
+        utils::SkipList<Vertex> &indexed_vertices);
 
     void LoadVerticesFromDiskLabelPropertyIndexForIntervalSearch(
-        LabelId label, PropertyId property, const std::unordered_set<storage::Gid> &gids,
+        const std::unordered_set<storage::Gid> &gids, LabelId label, PropertyId property,
         const std::optional<utils::Bound<PropertyValue>> &lower_bound,
         const std::optional<utils::Bound<PropertyValue>> &upper_bound, std::list<Delta> &index_deltas,
-        utils::SkipList<Vertex> *indexed_vertices);
+        utils::SkipList<Vertex> &indexed_vertices);
+
+    template <typename TIndex, typename TIndexCacheKey, typename TMergeFunc>
+    void SyncDeletedVertices(TIndex &index, TIndexCacheKey &cache_key,
+                             std::vector<std::list<Delta>> &index_delta_storage, View view, TMergeFunc &merge_func) {
+      if (!transaction_.vertices_to_delete_.empty()) {
+        index[cache_key] = utils::SkipList<Vertex>();
+        if (view == View::OLD) {
+          merge_func(index, index_delta_storage);
+        }
+      }
```

Comment:
```suggestion
      if (transaction_.vertices_to_delete_.empty()) {
        continue;
      }
      index[cache_key] = utils::SkipList<Vertex>();
      if (view == View::OLD) {
        merge_func(index, index_delta_storage);
      }
```
It's just a bit nicer with less indentation